File: Sem 4/Flutter/Lab 4/prime.dart
import 'dart:io';

int prime(int n) {
  bool flag = false;
  for (var i = 2; i <= n / 2; i++) {
    if (n % i == 0) {
      flag = true;
      break;
    }
  }
  if (flag)
    return 0;
  else
    return 1;
}

void main() {
  stdout.write("Enter num : ");
  int n = int.parse(stdin.readLineSync()!);

  print(prime(n));
}

File: Sem 4/Daa/Lab 2/prime.c
#include<stdio.h>
#include<time.h>
#include<stdbool.h>

bool prime(int n, int i) {
    if (n % i == 0 && i <=n/2 ) {
        return false;
    } else if (i <= 2) {
        return true;
    }
    prime(n,i+1);
}

void main() {
    int n , flag = 0;
    printf("Enter num : ");
    scanf("%d",&n);
    for (int i = 2; i <= n/2; i++) {
        if (n % i == 0) {
            flag = 1;   
            break;
        }
    }
    if (flag == 0) printf("prime");
    else printf("not prime");
    
    printf("%u",prime(n,2));
    

    
}